Simplified/NYPLBook.m

```diff
@@ -341,6 +341,8 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
 
     NSString *const genericString = NYPLNullToNil(dictionary[DeprecatedAcquisitionKey][@"generic"]);
     NSURL *const genericURL = genericString ? [NSURL URLWithString:genericString] : nil;
+      
+      // We should create a method that returns NYPLOPDSAcquisition object from NYPLOPDSAcquisitionRelation and NSURL object? Far less code and easier on the eyes.
```

Comment:
not sure I understand: isn't that what the `NYPLOPDSAcquisition::acquisitionWithRelation:type:hrefURL:indirectAcquisitions:availability:` method does?

Reply:
Agreed. I was thinking of creating another function that further simplifies it but that will create confusion. Leaving it unchanged

Reply:
However we'll have to refactor this method at some point since it's 220 lines long

Reply:
I fully agree on that. If you want to take a stab at it in lack of other tasks, I won't say no!